### PR TITLE
Nodejs v18 compatibility

### PR DESCRIPTION
**Pull request by pawelziebaopti**
This change is [Reviewable](https://reviewable.io/reviews/zaiusinc/node-sdk/54)

### CODE REVIEW

**Review by vickrum — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -24,34 +24,42 @@
...
-    "@types/deep-freeze": "^0.1.1",
-    "@types/jest": "^24.0.11",
-    "@types/nock": "^11.1.0",
-    "@types/node": "^12.6.9",
-    "@types/node-fetch": "^2.3.7",
-    "@zaiusinc/tslint-presets": "^1.0.1",
+    "@types/deep-freeze": "^0.1.2",
+    "@types/jest": "^28.1.8",
+    "@types/node": "^18.7.13",
+    "@types/node-fetch": "^2.6.2",
```
Comment (SoAG) [inline package.json:42]: What was the decision on upgrading to `v3.2.10`?

Comment (pawelziebaopti) [reply package.json:42]: there was no decision. to upgrade to v3.2.10 we would have to change packaging to EMS, which I'm not sure we can do. After reading about it, I think we should be fine to change the packaging, but I don't know enough about js world to be sure. 
if you're fine changing it, I'll give it a go

Comment (pawelziebaopti) [reply package.json:42]: one drawback with ESM is that jest has only experimental support for it https://jestjs.io/docs/ecmascript-modules

Comment (steelThread) [reply package.json:42]: @pawelziebaopti  - i thought the decision was to ignore the js-doc for the blank line between the comment and the args/return?  this should be consistently formatted across all our ts projects and it makes the diff unnecessarily noisy.

Comment (steelThread) [reply package.json:42]: i added a comment in the our eng team for the ESM switch.  i believe that will be a rather big change and not with the risk at this point in time just to support v3 1`node-fetch`.

Comment (pawelziebaopti) [reply package.json:42]: true, I upgraded eslint presets version and removed formatting changes

**Review by steelThread — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/zaiusinc/node-sdk/54)* status: 0 of 28 files reviewed, 2 unresolved discussions (waiting on @pawelziebaopti, @SoAG, and @steelThread)

___
*[`package.json` line 33 at r3](https://reviewable.io/reviews/zaiusinc/node-sdk/54#-NBqqnEy3nKs4eitGkZO:-NBqqnEy3nKs4eitGkZP:b8sp3q5) ([raw file](https://github.com/zaiusinc/node-sdk/blob/b3e92e5f32522ef0d22a6c2f3ce15342af9b39de/package.json#L33)):*
> ```JSON
>   },
>   "engines": {
>     "node": ">=18.0"
> ```

this package is a publicly accessible and usage able package independent of OCP.  was there any dependency on node 18 for any of the packages you updated?  i think we should remain backwards compatible for this one if we can.  you can test this by building the project with a node 12 version (8 is a little silly)

**Review by steelThread — COMMENTED** [review]
I noticed a couple of things:

1.  `ts-jest@28.0.8" has incorrect peer dependency "jest@^28.0.0"` - you need to update the jest version to match what ts-jest defines as a peer dependency.  also noticed ts-jest 29.x was just released.  lol
2.  `error eslint-plugin-jsdoc@39.3.6: The engine "node" is incompatible with this module. Expected version "^14 || ^16 || ^17 || ^18". Got "12.22.6"`.  the engine has to be >= 14 but only for development.  since 12 is not supported i'm thinking we should baseline the eng at 14 also although we could probably stay at 12.  thoughts?

> *[Reviewable](https://reviewable.io/reviews/zaiusinc/node-sdk/54#-:-NBrGKVG5iYdgA8mz7f-:brmvvnb)* status: 0 of 28 files reviewed, 2 unresolved discussions (waiting on @pawelziebaopti, @SoAG, and @steelThread)

**Review by steelThread — APPROVED** [review]
:lgtm: 

> Reviewed 8 of 28 files at r1, 1 of 3 files at r2, 17 of 18 files at r3, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/zaiusinc/node-sdk/54#-:-NBwQDg-EAv1diNtqBHx:bg7igkv)* status: all files reviewed, 1 unresolved discussion (waiting on @SoAG)

**Review by pawelziebaopti — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/zaiusinc/node-sdk/54)* status: all files reviewed, 1 unresolved discussion (waiting on @SoAG)

___
*[`package.json` line 42 at r1](https://reviewable.io/reviews/zaiusinc/node-sdk/54#-NAcdunRHLKsHWuQ5F_3-r1-42:-NBwVv8rAzENdz578b3l:bb7f2mf) ([raw file](https://github.com/zaiusinc/node-sdk/blob/b5348d449b3c690aa3f8ea307bd2650cfa4b2f69/package.json#L42)):*
<details><summary><i>Previously, pawelziebaopti (Pawel Zieba) wrote…</i></summary><blockquote>

true, I upgraded eslint presets version and removed formatting changes
</blockquote></details>

done

**Review by steelThread — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/zaiusinc/node-sdk/54)* status: all files reviewed, 1 unresolved discussion (waiting on @SoAG)

**Review by pawelziebaopti — COMMENTED** [review]
> Reviewed 8 of 28 files at r1, 1 of 3 files at r2, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/zaiusinc/node-sdk/54)* status: all files reviewed, 1 unresolved discussion (waiting on @SoAG)

**Pull request merged**